src/dmd/expressionsem.d

```diff
@@ -6715,6 +6715,42 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             return setError();
         }
 
+        if (tob.ty == Tarray && t1b.ty == Tarray && exp.e1.op != TOK.string_)
+        {
+            auto tFrom = t1b.nextOf();
+            auto tTo = tob.nextOf();
+
+            uint fromSize = cast(uint)tFrom.size();
+            uint toSize = cast(uint)tTo.size();
+
+            // If array element sizes do not match, we must adjust the dimensions
+            if (fromSize != toSize)
+            {
+                if ((fromSize % toSize) != 0)
+                {
+                    // A runtime check is needed in case arrays don't line up.  That check should
+                    // be done in the implementation of `object.__ArrayCast`
+
+                    // lower to `object.__ArrayCast!(TFrom, TTo)(from)`
+
+                    auto id = Identifier.idPool("__ArrayCast");
```

Comment:
should be `Identifier.__ArrayCast`, putting it in the `dmd.id.msgtable` will do that automatically.

Reply:
I did put it in `msgtable`, but if I use `Identifier.__ArrayCast` I get `Error: no property __ArrayCast for type dmd.identifier.Identifier`.  What am I doing wrong?

Reply:
Derp, that should be `Id.__ArrayCast`.